src/librustc_target/abi/mod.rs

```diff
@@ -782,6 +782,18 @@ pub enum Abi {
     Aggregate {
         /// If true, the size is exact, otherwise it's only a lower bound.
         sized: bool,
+
+        /// If true, then this is a `#[repr(C)]` aggregate which
+        /// contains a "Variable Length Array" -- or something which
+        /// appears to be a VLA -- in the final field. Such arrays are
+        /// treated specially by some ABIs. In particular, the
+        /// "homogeneous aggregate" test largely ignores zero-sized
+        /// types, with the exception that a VLA-like struct is never
+        /// a homogeneous aggregate.
+        ///
+        /// (Note that this field is always false for things that are
+        /// not `#[repr(C)]`.)
```

Comment:
> Note that this field is always false for things that are not `#[repr(C)]`

Seems like a difficult requirement to uphold in the future without something asserting this somewhere. I feel like the code downstream should be able to handle this field regardless of the overall ABI (i.e. interpret it semantically).

---

If an aggregate has a VLA within itself, it itself becomes variable-length, in which case it would make sense to just name this field as `is_variable_length`... 

Which then brings up a question: does it make sense for a variable-length aggregate to also be `sized`? Well, at least the `#[repr(C)] struct Foo { last: [T; 0] }` would be `Aggregate { sized: true, variable_length: true }`, but then what really is it – a variable length aggregate or a zero-sized aggregate? Hmm...

To be fair, I’m not sure what ABI interactions arise here – the currently implemented behaviour may well be the correct one...